src/lib/Core.hs

```diff
@@ -269,6 +270,20 @@ instance (Monad m, ExtOutMap Env decls, OutFrag decls)
           return (ans, catOutFrags (toScope env') decls d, env')
   {-# INLINE refreshAbs #-}
 
+instance ( Monad m, ExtOutMap Env d1, ExtOutMap Env d2
+         , OutFrag d1, OutFrag d2, SubstB Name d1, HoistableB d1)
+         => EnvExtender (DoubleInplaceT Env d1 d2 m) where
```

Comment:
How come we didn't need this before?

Reply:
Before we only used `DoubleInplaceT` in `DoubleBuilder`, and we implemented the `EnvExtender` instance directly.